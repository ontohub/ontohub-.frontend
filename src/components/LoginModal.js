import React, { Component } from 'react'
import { Button, Dimmer, Grid, Header, Loader, Modal } from 'semantic-ui-react'
import SignInForm from './SignInForm'
import SignUpForm from './SignUpForm'

export class LoginModal extends Component {
  constructor(props) {
    super(props)
    this.state = {
      open: this.props.open,
      loading: false
    }
    this.onSignInSubmit = this.onSignInSubmit.bind(this)
    this.onSignUpSubmit = this.onSignUpSubmit.bind(this)
    this.onError = this.onError.bind(this)
    this.onClose = this.onClose.bind(this)
    this.onOpen = this.onOpen.bind(this)
  }
  onSignInSubmit(username, password) {
    this.setState({ loading: true })
    return this.props.onSignIn(username, password)
  }
  onSignUpSubmit(username, email, password, captcha) {
    this.setState({ loading: true })
    return this.props.onSignUp(username, email, password, captcha)
  }
  onError() {
    this.setState({ loading: false })
  }
  onClose() {
    this.setState({ open: false })
  }
  onOpen() {
    this.setState({ open: true })
  }
  render() {
    return (
      <Modal
        size="small"
        open={this.state.open}
        closeIcon
        style={{ overflow: 'hidden' }}
        trigger={
          <Button
            primary
            icon="sign in"
            id="login-modal-sign-in-button"
            content="Sign in"
          />
        }
      >
        <Modal.Header>Sign in</Modal.Header>
        <Modal.Content>
          <Dimmer inverted active={this.state.loading}>
            <Loader />
          </Dimmer>
          <Grid celled="internally" stackable divided>
            <Grid.Row>
              <Grid.Column>
                <SignInForm
                  onSubmit={this.onSignInSubmit}
                  onSuccess={this.onClose}
                  onError={this.onError}
                />
              </Grid.Column>
            </Grid.Row>
            <Grid.Row>
              <Grid.Column>
                <Header textAlign="center">
                  Don't have an account yet? Sign up now!
                </Header>
                <SignUpForm
                  enableCaptcha={this.props.enableCaptcha}
                  validations={this.props.signUpValidations}
                  onSubmit={this.onSignUpSubmit}
                  onSuccess={this.onClose}
                  onError={this.onError}
                />
              </Grid.Column>
            </Grid.Row>
          </Grid>
        </Modal.Content>
      </Modal>
    )
  }
}

export default LoginModal
