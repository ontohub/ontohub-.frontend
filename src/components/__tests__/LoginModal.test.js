import React from "react";
import LoginModal from "../LoginModal";

describe("LoginModal", () => {
  let wrapper, onSignIn, onSignUp;

  beforeEach(() => {
    onSignIn = jest.fn();
    onSignUp = jest.fn();
    wrapper = mount(
      <LoginModal open onSignIn={onSignIn} onSignUp={onSignUp} />
    );
  });

  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("Callbacks", () => {
    describe("onSignInSubmit", () => {
      it("sets the state to loading", () => {
        wrapper.instance().onSignInSubmit();
        expect(wrapper.state("loading")).toBeTruthy();
      });

      it("calls the passed function", () => {
        wrapper.instance().onSignInSubmit();
        expect(onSignIn.mock.calls.length).toBe(1);
      });
    });
    describe("onSignUpSubmit", () => {
      it("sets the state to loading", () => {
        wrapper.instance().onSignUpSubmit({});
        expect(wrapper.state("loading")).toBeTruthy();
      });

      it("calls the passed function", () => {
        wrapper.instance().onSignUpSubmit({});
        expect(onSignUp.mock.calls.length).toBe(1);
      });
    });
    describe("onError", () => {
      it("removes the loading state", () => {
        wrapper.instance().onError();
        expect(wrapper.state("loading")).toBeFalsy();
      });
    });
    describe("onClose", () => {
      it("sets the open state to false", () => {
        wrapper.instance().onClose();
        expect(wrapper.state("open")).toBeFalsy();
      });
    });
    describe("onClose", () => {
      it("sets the open state to true", () => {
        wrapper.instance().onOpen();
        expect(wrapper.state("open")).toBeTruthy();
      });
    });
  });
});
