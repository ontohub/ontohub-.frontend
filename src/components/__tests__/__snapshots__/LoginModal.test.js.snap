// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal matches the snapshot 1`] = `
<LoginModal
  onSignIn={[Function]}
  onSignUp={[Function]}
  open={true}
>
  <Modal
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    open={true}
    size="small"
    style={
      Object {
        "overflow": "hidden",
      }
    }
    trigger={
      <Button
        as="button"
        content="Sign in"
        icon="sign in"
        id="login-modal-sign-in-button"
        primary={true}
      />
    }
  >
    <Portal
      className="ui page modals dimmer transition visible active"
      closeOnDocumentClick={false}
      closeOnEscape={true}
      closeOnRootNodeClick={true}
      mountNode={
        <body
          class="dimmable dimmed"
        >
          <div
            class="ui page modals dimmer transition visible active"
          >
            <div
              data-reactroot=""
              class="ui small modal transition visible active"
              style="overflow: hidden;"
            >
              <i
                aria-hidden="true"
                class="close icon"
              />
              <div
                class="header"
              >
                Sign in
              </div>
              <div
                class="content"
              >
                <div
                  class="ui inverted dimmer"
                >
                  <div
                    class="content"
                  >
                    <div
                      class="center"
                    >
                      <div
                        class="ui loader"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="ui stackable internally celled divided grid"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="column"
                    >
                      <form
                        class="ui form"
                      >
                        <div
                          class="equal width fields"
                        >
                          <div
                            class="field"
                          >
                            <label
                              for="sign-in-username"
                            >
                              Username
                            </label>
                            <input
                              placeholder="Username"
                              id="sign-in-username"
                              value="ada"
                            />
                          </div>
                          <div
                            class="field"
                          >
                            <label
                              for="sign-in-password"
                            >
                              Password
                            </label>
                            <input
                              type="password"
                              placeholder="Password"
                              id="sign-in-password"
                              value="changemenow"
                            />
                          </div>
                        </div>
                        <button
                          type="submit"
                          id="sign-in-form-sign-in-button"
                          class="ui primary right floated button"
                          role="button"
                        >
                          <i
                            aria-hidden="true"
                            class="sign in icon"
                          />
                          <!-- react-text: 22 -->
                          Sign in
                          <!-- /react-text -->
                        </button>
                      </form>
                    </div>
                  </div>
                  <div
                    class="row"
                  >
                    <div
                      class="column"
                    >
                      <div
                        class="ui center aligned header"
                      >
                        Don't have an account yet? Sign up now!
                      </div>
                      <form
                        class="ui form"
                      >
                        <div
                          class="equal width fields"
                        >
                          <div
                            class="field"
                          >
                            <label
                              for="sign-up-username"
                            >
                              Username
                            </label>
                            <input
                              placeholder="Username"
                              id="sign-up-username"
                            />
                          </div>
                          <div
                            class="field"
                          >
                            <label
                              for="sign-up-email"
                            >
                              Email
                            </label>
                            <input
                              placeholder="Email"
                              id="sign-up-email"
                            />
                          </div>
                        </div>
                        <div
                          class="equal width fields"
                        >
                          <div
                            class="field"
                          >
                            <label
                              for="sign-up-password"
                            >
                              Password
                            </label>
                            <input
                              type="password"
                              placeholder="Password"
                              id="sign-up-password"
                            />
                            <div
                              title="Password strength: Very weak"
                              class="ui red bottom attached progress"
                              data-percent="0"
                              style="margin-top: -3px;"
                            >
                              <div
                                class="bar"
                                style="width: 0%;"
                              />
                            </div>
                          </div>
                          <div
                            class="field"
                          >
                            <label
                              for="sign-up-password-confirm"
                            >
                              Confirm Password
                            </label>
                            <input
                              type="password"
                              placeholder="Confirm Password"
                              id="sign-up-password-confirm"
                            />
                          </div>
                        </div>
                        <button
                          type="submit"
                          class="ui basic secondary right floated button"
                          role="button"
                        >
                          <i
                            aria-hidden="true"
                            class="add user icon"
                          />
                          <!-- react-text: 45 -->
                          Create account
                          <!-- /react-text -->
                        </button>
                      </form>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </body>
      }
      onClose={[Function]}
      onMount={[Function]}
      onOpen={[Function]}
      onUnmount={[Function]}
      open={true}
      openOnTriggerClick={true}
      trigger={
        <Button
          as="button"
          content="Sign in"
          icon="sign in"
          id="login-modal-sign-in-button"
          primary={true}
        />
      }
    >
      <Button
        as="button"
        content="Sign in"
        icon="sign in"
        id="login-modal-sign-in-button"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        primary={true}
      >
        <button
          className="ui primary button"
          id="login-modal-sign-in-button"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
        >
          <Icon
            as="i"
            name="sign in"
          >
            <i
              aria-hidden="true"
              className="sign in icon"
            />
          </Icon>
          Sign in
        </button>
      </Button>
    </Portal>
  </Modal>
</LoginModal>
`;
